lib/generators/doorkeeper/openid_connect/templates/initializer.rb

```diff
@@ -24,9 +26,12 @@
     # redirect_to new_user_session_url
   end
 
-  subject do |resource_owner|
+  subject do |resource_owner, application|
     # Example implementation:
-    # resource_owner.key
+    # resource_owner.id
+
+    # or if you need pairwise subject identifier, implement like below:
+    # Digest::SHA256.hexdigest("#{resource_owner.id}#{URI.parse(application.redirect_uri).host}#{'your_secret_salt'}")
```

Comment:
I don't think it's a good idea to reuse the same lambda for the pairwise algorithm, and 'm wondering if we need to support both pairwise and non-pairwise algorithms at the same time.

From the [specs](http://openid.net/specs/openid-connect-core-1_0.html#SubjectIDTypes):

> The OpenID Provider's Discovery document SHOULD list its supported Subject Identifier types in the `subject_types_supported` element. If there is more than one type listed in the array, the Client MAY elect to provide its preferred identifier type using the `subject_type` parameter during Registration. 

If I understand this correctly, the `subject_type` is entirely determined by the server side and can't be requested by the client, and we currently don't support dynamic registration or have a `subject_type` attribute on `oauth_applications` so we don't necessarily need to support both simultaneously, but it might be better in preperation to adding registration in the future.

What do you think about adding a second setting called `pairwise_subject`, and then make `subject_types_supported` dynamic, i.e. check if `subject` and `pairwise_subject` are defined and return the corresponding keys?